extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php

```diff
@@ -39,20 +39,46 @@ public function getAssets() {
 	}
 
 	public function publish() {
-		$response = $this->getResponse();
-		$response->setFormat( WikiaResponse::FORMAT_JSON );
-
-		$status = $this->attemptSave( $this->getRequest()->getParams() );
-
 		$requestParams = $this->getRequest()->getParams();
+		$status = $this->attemptSave( $requestParams );
+
 		$urls = PortableInfoboxBuilderHelper::createRedirectUrls( $requestParams['title'] );
 
+		$response = $this->getResponse();
+		$response->setFormat( WikiaResponse::FORMAT_JSON );
 		$response->setVal( 'urls', $urls );
 		$response->setVal( 'success', $status->isOK() );
 		$response->setVal( 'errors', $status->getErrorsArray() );
 		$response->setVal( 'warnings', $status->getWarningsArray() );
 	}
 
+	public function publishAndRename() {
```

Comment:
I know we didn't do that in the publish() method, yet I think it's high time to describe the expected request params to make sure it's clear what we're receiving.